File: packages/extism-tools/examples/wasm/wiki.js
import createPlugin from "@extism/extism";

async function main() {
  const plugin = await createPlugin("./dist/wasm/wiki.wasm", {
    useWasi: true,
    functions: {},
    runInWorker: true,
    allowedHosts: ["*.wikipedia.org"],
    memory: { maxHttpResponseBytes: 100 * 1024 * 1024 },
  });

  try {
    const params = { query: "Ho Chi Minh City" };
    const data = await plugin.call("summary", JSON.stringify(params));
    console.log(data.json());
  } catch (e) {
    console.error(e);
  } finally {
    await plugin.close();
  }
}

void main();

File: packages/extism-tools/examples/src/wiki.py
import extism
import json
from os.path import join, dirname


def read_local_wasm(file_name):
    path = join(dirname(__file__), file_name)  # Change this to your wasm file path
    with open(path, "rb") as wasm_file:
        return wasm_file.read()


def _manifest(file_name):
    wasm = read_local_wasm(file_name)
    return {
        "wasm": [{"data": wasm}],
        "allowed_hosts": ["*.wikipedia.org"],
    }


manifest = _manifest("wiki.wasm")
with extism.Plugin(manifest, wasi=True) as plugin:
    metadata = plugin.call(
        "getMetadata",
        "",
        parse=lambda output: json.loads(bytes(output).decode("utf-8")),
    )
    data = plugin.call(
        "call",
        json.dumps({"query": "Ho Chi Minh City"}),
        parse=lambda output: json.loads(bytes(output).decode("utf-8")),
    )

print(metadata)
print(data)
